openwisp_notifications/tests/test_notifications.py

```diff
@@ -1052,6 +1053,24 @@ def test_that_the_notification_is_only_sent_once_to_the_user(self):
         self._create_notification()
         self.assertEqual(notification_queryset.count(), 1)
 
+    def test_email_unsubscribe_token(self):
```

Comment:
Add tests for the whole view. That's why the coverage is falling. I would like to see test with the following scenarios: 

- token not present in the URL 
- token is malformed (error occurs on base64.decode and JSON cannot be parsed after decoding)
- token is invalid for the user present in the JSON
- user does not exist for the passed UUID
- token is valid and the page is rendered properly with the option to change preference, then make a POST request to the endpoint and verify notifications has been unsubscribed.

Reply:
I have added the tests 

https://github.com/openwisp/openwisp-notifications/pull/307/commits/735c4d0518d8a2bd8a9f2e528ee3ef07e1cbe56c#diff-3598a9f8ba373465969faa6e1fa78c442a295198b7e4bf3ba004d7a94d3a3d80R1076-R1080